### ISSUE TITLE
Debug: Support for breakpoints in libraries 

### ISSUE THREAD

**Issue by frisitano** [0xMiden/miden-vm#779]
When developing a library it would be useful to be able to place breakpoints within library procedures. Currently when I try and place a breakpoint in a library procedure and load the library via the debugger (using #776) I get the following error:
```
============================================================
Debug program
============================================================
Reading Library file `../miden-base/miden-lib/assets/miden.masl`
Failed to read library! - unexpected reader EOF
```

**Comment by vlopes11** [comment]
We currently don't support breakpoint for libraries. The breakpoint instruction is just a quick workaround until we have our source mapping implemented.

We *can* support such breakpoint for libraries, but then we would change their code in production. If we opt to go that way, we need to serialize the opcode here

https://github.com/0xPolygonMiden/miden-vm/blob/b1cfe793a6db5b9e4aea543b0e2547d37d97cf02/assembly/src/parsers/serde/serialization.rs#L471-L474

**Comment by bitwalker** [comment]
I think this will be best fixed by making `midenc-debug` the underlying debugger of the Miden CLI. It supports breakpoints in any part of the program, and has a variety of breakpoint types as well, including the ability to specify them by source file/line, entry to a procedure, etc.

**Comment by bobbinth** [comment]
@bitwalker - I think you recently integrated `breakpoint` instruction into your debugger. Does it also handle breakpoints in libraries? If so, we can probably close this issue.

**Comment by bitwalker** [comment]
Yes, it supports breakpoints anywhere, with a handful of options for setting them, so I think this can be closed.
